src/wasm/wasm-stack.cpp

```diff
@@ -1982,19 +1982,21 @@ void BinaryInstWriter::visitRefCast(RefCast* curr) {
 }
 
 void BinaryInstWriter::visitBrOn(BrOn* curr) {
-  o << int8_t(BinaryConsts::GCPrefix);
   switch (curr->op) {
+    case BrOnNull:
+      o << int8_t(BinaryConsts::BrOnNull);
+      break;
     case BrOnCast:
-      o << U32LEB(BinaryConsts::BrOnCast);
+      o << int8_t(BinaryConsts::GCPrefix) << int8_t(BinaryConsts::BrOnCast);
```

Comment:
an `int8_t` must be used above for `BrOnNull`, or else as the opcode is large enough it will be emitted with an extra 00 byte after as an LEB. I made the same change for the others as well - I think this was a bug there before. And it now matches RefAs.

Reply:
This is correct for `BrOnNull` because un-prefixed opcodes are bytes, but it is not correct to make this change for the other instructions. All prefixed opcodes are LEBs.

Reply:
Thanks for the clarification, that wasn't clear to me from the spec.